### DISCUSSION TITLE
FastAPI SQLModel not working

### DISCUSSION THREAD

**Discussion by sandchak** [fastapi/fastapi#12953] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Depends
from sqlmodel import SQLModel, Field, create_engine, Session
from typing import Annotated
from enum import Enum
from contextlib import asynccontextmanager

sqllite_url = "sqlite:///database.db"
db_connect_args = {"check_same_thread": False}
engine = create_engine(sqllite_url, connect_args=db_connect_args)

class Sex(Enum):
    Male = 'Male'
    Female = 'Female'
    Other = 'Other'

class Student(SQLModel, Table=True):
    id: int | None = Field(default=None, primary_key=True)
    first_name: str 
    middle_name: str | None = Field(default=None)
    last_name: str
    full_name: str = Field(index=True)
    age: int 
    sex: Sex
    standard: str = Field(index=True)
    section: str = Field(index=True)

sandip = Student(first_name='sandip', last_name='chakraborty', full_name='sandip chakraborty', age=39, sex=Sex.Male, standard='10', section='A')

@asynccontextmanager
async def create_db_and_tables(app: FastAPI):
    print("On Startup.....")
    SQLModel.metadata.create_all(engine)
    with Session(engine, expire_on_commit=False) as session:
        session.add(sandip)
        session.commit()
        print(session.get(Student, 1))
    yield
    SQLModel.metadata.drop_all(engine)


app = FastAPI(lifespan=create_db_and_tables)

def create_session():
    with Session(engine, expire_on_commit=False) as session:
        yield session

session_dep = Annotated[Session, Depends(create_session)]

@app.post("/students")
def add_student(student: Student, session: session_dep) -> Student:
    print(student)
    session.add(student)
    session.commit()
    session.refresh(student)
    print(student)
    return student
```

### Description

On start I am trying to save one SQLModel object to DB but it is giving below error...

PS C:\Users\echysap\Documents\python_fastapi\database> uvicorn main:app --reload
INFO:     Will watch for changes in these directories: ['C:\\Users\\echysap\\Documents\\python_fastapi\\database']
INFO:     Uvicorn running on http://127.0.0.1:8000 (Press CTRL+C to quit)
INFO:     Started reloader process [19576] using StatReload
INFO:     Started server process [17760]
INFO:     Waiting for application startup.
On Startup.....
ERROR:    Traceback (most recent call last):
  File "C:\python\Python310\lib\site-packages\sqlalchemy\orm\session.py", line 3473, in add
    state = attributes.instance_state(instance)
  File "C:\python\Python310\lib\site-packages\pydantic\main.py", line 856, in __getattr__
    raise AttributeError(f'{type(self).__name__!r} object has no attribute {item!r}')
AttributeError: 'Student' object has no attribute '_sa_instance_state'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\python\Python310\lib\site-packages\starlette\routing.py", line 693, in lifespan
    async with self.lifespan_context(app) as maybe_state:
  File "C:\python\Python310\lib\contextlib.py", line 199, in __aenter__
    return await anext(self.gen)
  File "C:\Users\echysap\Documents\python_fastapi\database\main.py", line 34, in create_db_and_tables
    session.add(sandip)
  File "C:\python\Python310\lib\site-packages\sqlalchemy\orm\session.py", line 3475, in add
    raise exc.UnmappedInstanceError(instance) from err
sqlalchemy.orm.exc.UnmappedInstanceError: Class 'main.Student' is not mapped

    raise exc.UnmappedInstanceError(instance) from err
sqlalchemy.orm.exc.UnmappedInstanceError: Class 'main.Student' is not mapped
    raise exc.UnmappedInstanceError(instance) from err
sqlalchemy.orm.exc.UnmappedInstanceError: Class 'main.Student' is not mapped
ERROR:    Application startup failed. Exiting.
INFO:     Stopping reloader process [19576]

### Operating System

Windows

### Operating System Details

Windows 10 

### FastAPI Version

0.115.2

### Pydantic Version

2.9.2

### Python Version

Python 3.10.11

### Additional Context

Remaining library versions are here
PS C:\Users\echysap\Documents\python_fastapi\database> pip freeze
annotated-types==0.7.0
anyio==4.6.2.post1
bcrypt==4.2.0
click==8.1.7
colorama==0.4.6
exceptiongroup==1.2.2
fastapi==0.115.2
greenlet==3.1.1
h11==0.14.0
idna==3.10
passlib==1.7.4
pydantic==2.9.2
pydantic_core==2.23.4
PyJWT==2.9.0
python-multipart==0.0.12
sniffio==1.3.1
SQLAlchemy==2.0.36
sqlmodel==0.0.22
starlette==0.40.0
typing_extensions==4.12.2
uvicorn==0.32.0
PS C:\Users\echysap\Documents\python_fastapi\database>

**Comment by YuriiMotov** [comment, marked as answer]
`class Student(SQLModel, Table=True):`
Here is an error. It should be `class Student(SQLModel, table=True):` ('table' all in lowercase)

**Comment by sandchak** [comment]
Thanks @YuriiMotov.. Silly mistake from me... :-)
